watertap/unit_models/nanofiltration_DSPMDE_0D.py

```diff
@@ -2015,6 +2021,139 @@ def calculate_scaling_factors(self):
             sf = iscale.get_scaling_factor(self.flux_mol_phase_comp[t, x, p, j])
             iscale.constraint_scaling_transform(con, sf)
 
+        for (t, x, p, j), con in self.eq_interfacial_partitioning_feed.items():
+            hfd = value(self.hindrance_factor_diffusive_comp[t, j])
+            bs = value(self.partition_factor_born_solvation_comp[t, j])
+            fdc = value(self.partition_factor_donnan_comp_feed[t, x, j])
+            hinderence_factor = hfd * bs * fdc
+            # we expect that there will be less ion at membrane with same charge as it self
+            # eg. negative ions ar repulsed by negative membarnes, but positve ions are attracted
+            membrane_charge = value(self.membrane_charge_density[0])
+            if (
+                membrane_charge > 0
+                and value(self.permeate_side[t, x].charge_comp[j]) > 0
+            ):
+                charge_reflection_multiplier = 10 ** -abs(
+                    value(self.permeate_side[t, x].charge_comp[j])
+                )
+            elif (
+                membrane_charge < 0
+                and value(self.permeate_side[t, x].charge_comp[j]) < 0
+            ):
+                charge_reflection_multiplier = 10 ** -abs(
+                    value(self.permeate_side[t, x].charge_comp[j])
+                )
+            else:
+                charge_reflection_multiplier = 10 ** abs(
+                    value(self.permeate_side[t, x].charge_comp[j])
+                )
```

Comment:
Very interesting. I have a couple of comments:
1) Did you find that `hindrance_factor` was insufficient and that also dividing by this `charge_reflection_multiplier` was necessary? Following the equation for interfacial partitioning, one would think the `charge_reflection_multiplier` wouldn't be necessary.
2) In the case that the membrane charge density were unfixed, the user would need to know to set to some guess value I suppose.
3) Assuming the usage of `charge_reflection_multiplier ` makes for better scaling and we keep the logic as is, I wonder if this can be condensed. It seems that you check whether ion and membrane have same charge (either positive or negative) or not. Given that, could we do the following instead?
```suggestion
            membrane_charge = value(self.membrane_charge_density[0])
            # multiply the two numbers. if one is different charge, outcome will be negative; if same charge, whether negative or positive, outcome will be positive.
            if (
                membrane_charge* value(self.permeate_side[t, x].charge_comp[j])  > 0
               
            ):
                charge_reflection_multiplier = 10 ** -abs(
                    value(self.permeate_side[t, x].charge_comp[j])
                )
           else:
                charge_reflection_multiplier = 10 ** abs(
                    value(self.permeate_side[t, x].charge_comp[j])
                )
```

Reply:
Yeah, I observed this empirically. You basically need to know apriori how ions will partition because we have not computed the ion concentration ratio between pore entrance and interface yet and thats what the charge ratio tries to establish. So I think we have to make assumptions of membrane charge direction, rather then magnitude here. I also really doubt a user would be able to optimize from negative to postive with out things exploding, what happens when charge cross zero? (I don't want to find out?) 

```
b.pore_entrance[t, x].act_coeff_phase_comp[p, j]
                * b.pore_entrance[t, x].conc_mol_phase_comp[p, j]
                / (
                    b.feed_side.properties_interface[t, x].act_coeff_phase_comp[p, j]
                    * b.feed_side.properties_interface[t, x].conc_mol_phase_comp[p, j]
                )
```

Reply:
Also, yes I will condese this.